runtime/lua/vim/lsp.lua

```diff
@@ -16,24 +7,33 @@ local tbl_isempty, tbl_extend = vim.tbl_isempty, vim.tbl_extend
 local validate = vim.validate
 local if_nil = vim.F.if_nil
 
-local lsp = {
-  protocol = protocol,
-
-  handlers = default_handlers,
-
-  buf = require('vim.lsp.buf'),
-  diagnostic = require('vim.lsp.diagnostic'),
-  codelens = require('vim.lsp.codelens'),
-  inlay_hint = require('vim.lsp.inlay_hint'),
-  semantic_tokens = semantic_tokens,
-  util = util,
+local lsp = vim._defer_require('vim.lsp', {
+  _changetracking = ..., --- @module 'vim.lsp._changetracking'
+  _completion = ..., --- @module 'vim.lsp._completion'
+  _dynamic = ..., --- @module 'vim.lsp._dynamic'
+  _snippet_grammar = ..., --- @module 'vim.lsp._snippet_grammar'
+  _watchfiles = ..., --- @module 'vim.lsp._watchfiles'
+  buf = ..., --- @module 'vim.lsp.buf'
+  codelens = ..., --- @module 'vim.lsp.codelens'
+  diagnostic = ..., --- @module 'vim.lsp.diagnostic'
+  handlers = ..., --- @module 'vim.lsp.handlers'
+  inlay_hint = ..., --- @module 'vim.lsp.inlay_hint'
+  log = ..., --- @module 'vim.lsp.log'
+  protocol = ..., --- @module 'vim.lsp.protocol'
+  rpc = ..., --- @module 'vim.lsp.rpc'
+  semantic_tokens = ..., --- @module 'vim.lsp.semantic_tokens'
+  tagfunc = ..., --- @module 'vim.lsp.tagfunc'
+  util = ..., --- @module 'vim.lsp.util'
+})
 
-  -- Allow raw RPC access.
-  rpc = lsp_rpc,
+local log = lsp.log
+local protocol = lsp.protocol
+local ms = protocol.Methods
+local util = lsp.util
+local changetracking = lsp._changetracking
 
-  -- Export these directly from rpc.
-  rpc_response_error = lsp_rpc.rpc_response_error,
-}
+-- Export these directly from rpc.
```

Comment:
This can be used to make the `TODO: documentation...` not appear.

```suggestion
-- Export these directly from rpc.
---@nodoc
```